Sources/Services/AWSACM/Plugins.swift

```diff
@@ -0,0 +1,59 @@
+// Code generated by smithy-swift-codegen. DO NOT EDIT!
+
+@_spi(FileBasedConfig) import AWSClientRuntime
+import ClientRuntime
+
+public class ACMClientEndpointPlugin: Plugin {
+    private var endpointResolver: EndpointResolver
+    public init(endpointResolver: EndpointResolver) {
+        self.endpointResolver = endpointResolver
+    }
+    public convenience init() throws {
+        self.init(endpointResolver: try DefaultEndpointResolver())
+    }
+    public func configureClient(clientConfiguration: ClientRuntime.ClientConfiguration) throws {
+        if var config = clientConfiguration as? ACMClient.ACMClientConfiguration {
+            config.endpointResolver = self.endpointResolver
+        }
+    }
+}
+
+public class DefaultAWSAuthSchemePlugin: Plugin {
+    private var authSchemes: [ClientRuntime.AuthScheme]?
+    private var authSchemeResolver: ClientRuntime.AuthSchemeResolver
+    public init() {
+        self.authSchemeResolver = DefaultACMAuthSchemeResolver()
+        self.authSchemes = [SigV4AuthScheme()]
+    }
+    public func configureClient(clientConfiguration: ClientRuntime.ClientConfiguration) throws {
+        if var config = clientConfiguration as? ACMClient.ACMClientConfiguration {
+            config.authSchemeResolver = self.authSchemeResolver
+            config.authSchemes = self.authSchemes
+            config.awsCredentialIdentityResolver = try AWSClientConfigDefaultsProvider.awsCredentialIdentityResolver()
+        }
+    }
+}
+
+public class ACMClientAuthSchemePlugin: Plugin {
+    private var authSchemes: [ClientRuntime.AuthScheme]?
+    private var authSchemeResolver: ClientRuntime.AuthSchemeResolver?
+    private var awsCredentialIdentityResolver: (any AWSClientRuntime.AWSCredentialIdentityResolver)?
+    public init(authSchemes: [ClientRuntime.AuthScheme]? = nil, authSchemeResolver: ACMAuthSchemeResolver? = nil, awsCredentialIdentityResolver: (any AWSClientRuntime.AWSCredentialIdentityResolver)? = nil) {
+        self.authSchemeResolver = authSchemeResolver
+        self.authSchemes = authSchemes
+        self.awsCredentialIdentityResolver = awsCredentialIdentityResolver
+    }
+    public func configureClient(clientConfiguration: ClientRuntime.ClientConfiguration) throws {
+        if var config = clientConfiguration as? ACMClient.ACMClientConfiguration {
+            if (self.authSchemes != nil) {
+                config.authSchemes = self.authSchemes
+            }
+            if (self.authSchemeResolver != nil) {
+                config.authSchemeResolver = self.authSchemeResolver!
+            }
+            if (self.awsCredentialIdentityResolver != nil) {
+                config.awsCredentialIdentityResolver = self.awsCredentialIdentityResolver!
+            }
+        }
+    }
+}
```

Comment:
Q: Why do we have two plugins for auth scheme related things? Could these maybe be combined into one plugin `ACMClientAuthSchemePlugin` by checking if set value is nil (like it's doing rn) and if it is nil, setting it to default value (like its done in `DefaultAWSAuthSchemePlugin`?